clients/da-vinci-client/src/main/java/com/linkedin/davinci/consumer/VersionSwapDataChangeListener.java

```diff
@@ -69,7 +71,11 @@ public void handleStoreChanged(Store store) {
 
         LOGGER.info(
             "New Version detected!  Seeking consumer to version: " + currentVersion + " in consumer: " + consumerName);
-        this.consumer.seekToEndOfPush(partitions).get();
+        this.consumer
+            .internalSeekToEndOfPush(
+                partitions,
+                pubSubTopicRepository.getTopic(store.getVersion(currentVersion).kafkaTopicName()))
+            .get();
       } catch (Exception e) {
         LOGGER.error(
```

Comment:
Should we try to do a bit more than just log the error here since there could be transient errors and we might not get another store change event for a while (maybe around the time when the currently subscribed VT is getting deleted). We could schedule or perform an immediate retry attempt to check for new current version and perform version swap?

Reply:
yeah let's do that